### PR TITLE
Implement hardware offloading override

### PR DESCRIPTION
**Pull request by phpb-com**
Implement hardware offloading override for selected interfaces. Allow to have global flag to turn-off all overrides. Also clean-up how hardware offloading is set/unset, since flags are idempotent.  #1431

### CODE REVIEW

File: src/etc/inc/interfaces.lib.inc [hunk 1]
```diff
@@ -297,38 +297,36 @@ function legacy_interface_details($intf)
...
-        if (isset($config['system']['disablechecksumoffloading']) && $csumv6_set) {
-            legacy_interface_flags($ifs, '-txcsum6 -rxcsum6', false);
-        } elseif (!isset($config['system']['disablechecksumoffloading']) && !$csumv6_set) {
-            legacy_interface_flags($ifs, 'txcsum6 rxcsum6', false);
+
+	$override = in_array($ifs, $overrideifs);
+
+        // hardware checksum offloading offloading
+        if (isset($config['system']['disablechecksumoffloading']) && !$override) {
+            legacy_interface_flags($ifs, '-txcsum -rxcsum -txcsum6 -rxcsum6', false);
```
Comment (fichtner) [inline src/etc/inc/interfaces.lib.inc]: we should merge this before we merge the feature

File: src/etc/inc/interfaces.lib.inc [hunk 2]
```diff
@@ -297,38 +297,36 @@ function legacy_interface_details($intf)
...
+            legacy_interface_flags($ifs, 'txcsum rxcsum txcsum6 rxcsum6', false);
         }
 
         // TCP segmentation offloading
-        if (isset($config['system']['disablesegmentationoffloading']) && $tso_set) {
-            legacy_interface_flags($ifs, '-tso', false);
-        } elseif (!isset($config['system']['disablesegmentationoffloading']) && !$tso_set) {
-            legacy_interface_flags($ifs, 'tso', false);
+        if (isset($config['system']['disablesegmentationoffloading']) && !$override) {
+            legacy_interface_flags($ifs, '-tso -tso4 -tso6', false);
```
Comment (fichtner) [inline src/etc/inc/interfaces.lib.inc]: also merge this earlier here (if it makes sense, but tso may include tso4 and tso6?)

File: src/etc/inc/interfaces.lib.inc [hunk 3]
```diff
@@ -297,38 +297,36 @@ function legacy_interface_details($intf)
...
+            $overrideifs = isset($config['system']['overridemembers']) ? explode(' ', $config['system']['overridemembers']) : array();
+        } else {
+            $overrideifs = array();
         }
-        if (isset($config['system']['disablechecksumoffloading']) && $csumv6_set) {
-            legacy_interface_flags($ifs, '-txcsum6 -rxcsum6', false);
-        } elseif (!isset($config['system']['disablechecksumoffloading']) && !$csumv6_set) {
-            legacy_interface_flags($ifs, 'txcsum6 rxcsum6', false);
+
+	$override = in_array($ifs, $overrideifs);
```
Comment (fichtner) [inline src/etc/inc/interfaces.lib.inc]: I think this merges logic for checkums and tso and lro, we shouldn't want that if we do it right

**Comment by phpb-com** [comment]
@fichtner thanks for review. I agree on all counts, will first retract the interface.lib.inc changes and submit as a separate PR. Then will change the override itself to per option for each interface. Thanks.

**Comment by phpb-com** [comment]
I will close this PR for now, since there is a need to fix the way flags are assigned to the interface, and the GUI in front of it.

**Pull request closed without merging**